tests/test_routes.py

```diff
@@ -0,0 +1,158 @@
+    )
+
+    if output_format == "google_sheet":
+        # assume get_manifest correctly returns a google sheet url
+        with patch.object(
+            ManifestGenerator, "get_manifest", return_value="google_sheet_url"
+        ) as mock_method:
+            result = mg.create_single_manifest(
+                access_token="mock_access_token",
+                single_data_type="Patient",
+                single_dataset_id="syn1234",
+            )
+            assert result == "google_sheet_url"
+    elif output_format == "dataframe":
+        # assume get_manifest correctly returns a dataframe
+        with patch.object(
+            ManifestGenerator, "get_manifest", return_value=pd.DataFrame()
+        ) as mock_method:
+            result = mg.create_single_manifest(
+                access_token="mock_access_token",
+                single_data_type="Patient",
+                single_dataset_id="syn1234",
+            )
+            assert isinstance(result, pd.DataFrame)
+
+
+@pytest.mark.parametrize("data_type", [["Patient, Biospecimen"], ["Patient"]])
+@pytest.mark.parametrize("output_format", ["google_sheet", "dataframe", "excel"])
+def test_generate_manifest_and_collect_outputs(
+    data_model_jsonld, output_format, caplog, data_type
+):
+    """Make sure that results get collected in a list when the output formats are google sheet and dataframe.
+    When output format is excel and there are multiple data types, make sure that a warning message gets triggered"""
+
+    mg = ManifestGeneration(
+        schema_url=data_model_jsonld, data_type=data_type, output_format=output_format
+    )
+    if output_format == "excel":
+        mock_return_val = "mock_excel_manifest"
+    elif output_format == "google_sheet":
+        mock_return_val = "mock_google_sheet_url"
+    else:
+        mock_return_val = pd.DataFrame({"col1": [1, 2], "col2": [3, 4]})
+
+    with patch.object(
+        ManifestGeneration, "create_single_manifest", return_value=mock_return_val
+    ) as mock_method:
+        result = mg.generate_manifest_and_collect_outputs(
+            access_token="mock_access_token",
+            data_type_lst=data_type,
+            dataset_id=["syn1234", "syn1235"],
+        )
+
+        # if the output is google sheet or dataframe, the number of items that get returned should match the number of items in data_type
+        if output_format != "excel":
+            assert len(result) == len(data_type)
+            assert isinstance(result, list)
+
+        else:
+            with caplog.at_level(logging.WARNING):
+                # return warning messages when there are multiple data types and the output format is excel
+                if len(data_type) > 1:
+                    assert (
+                        f"Currently we do not support returning multiple files as Excel format at once. Only manifest generated by using dataset id {data_type[0]} would get returned with title Example.{data_type[0]}.manifest"
```

Comment:
I would avoid using detailed conditionals like this because you're testing an implementation instead of an interface. If you change one word in your warning, you will cause the test to fail, which shouldn't happen if the interface hasn't changed.